diatheke/streams.py

```diff
@@ -0,0 +1,282 @@
+        if request is None:
+            # This is our sentinel to indicate the iterator should finish,
+            # which is done by raising the StopIteration exception. A side
+            # effect of this is that if a user sends 'None' to one of the
+            # send functions, it will also end the stream.
+            raise StopIteration
+
+        return request
+
+    def _run_stream(self, client_stub):
+        # Give the stream as the iterable the gRPC function requires.
+        result = client_stub.StreamASR(self)
+        self._set_result(result)
+
+    def _has_result(self):
+        with self._lock:
+            return self._result != None
+
+    def _set_result(self, result):
+        with self._lock:
+            self._result = result
+
+    def send_audio(self, audio_bytes):
+        """Send the given audio bytes to Diatheke for transcription.
+
+        If this function returns False, the server has closed the stream
+        and result() should be called to get the final ASR result."""
+        # Check if there is a result yet
+        if self._has_result():
+            return False
+
+        # Send the request
+        req = StreamASRRequest(audio=audio_bytes)
+        self._data_queue.put(req)
+        return True
+
+    def send_token(self, token):
+        """Send the given session token to Diatheke to update the
+        speech recognition context. The session token must first be
+        sent on the ASR stream before any audio will be recognized.
+        If the stream was created using client.new_session_asr_stream(),
+        the first token was already sent.
+
+        If this function returns False, the server has closed the stream
+        and result() should be called to get the final ASR result."""
+        # Check if there is a result yet
+        if self._has_result():
+            return False
+
+        # Send the request
+        req = StreamASRRequest(token=token)
+        self._data_queue.put(req)
+        return True
+
+    def result(self):
+        """Returns the result of speech recognition. This function may
+        be called to end the audio stream early, which will force a
+        transcription based on the audio received until this point, or
+        in response to receiving False from send_audio() or send_token()."""
+        # Check if the stream has already returned a result. If not,
+        # force it to stop by sending our sentinel iterator value.
+        if not self._has_result():
+            self._data_queue.put(None)
+
+        # Wait for the streaming thread to finish and return the result.
+        self._stream_thread.join()
+        return self._result
+
+
+def read_ASR_audio(stream, reader, buffSize):
+    """Deprecated. Use client.read_asr_audio() instead."""
+    # Check if we have a text reader
+    if isinstance(reader, io.TextIOBase):
+        is_text = True
+    else:
+        # Otherwise we assume we have a byte reader
+        is_text = False
```

Comment:
```suggestion
    # Assume we have a byte reader
    is_text = False
    # assign text reader if it's that
    if isinstance(reader, io.TextIOBase):
        is_text = True
```

Whatever one is the most expected/common should be set and then the if statement can check on the alternative case. 

Mostly a style/readability suggestion/